modules/dynamic-plugins/proc-extending-internal-icon-catalog.adoc

```diff
@@ -0,0 +1,27 @@
+[id="proc-extending-internal-icon-catalog"]
+
+= Extending internal icon catalog
+
+You can use the internal catalog to fetch icons for configured routes with sidebar navigation menu entry.
+
+.Procedure
+Add a custom icon to the internal icon catalog for use in a plugin's menu item by using the `appIcons` configuration as shown in the following example:
+
+[source,yaml]
+----
+# dynamic-plugins-config.yaml
+plugins:
+  - plugin: <plugin_path_or_url>
+    disabled: false
+    pluginConfig:
+      dynamicPlugins:
+        frontend:
+          <package_name>: # same as `scalprum.name` key in plugin's `package.json`
```

Comment:
This one comment in the YAML is actually a really important detail that we should highlight in the docs.  Also probably this example shouldn't include this comment.  However again folks often are confused where the "package_name" value comes from, it's whatever "scalprum.name" is set to in the package.json, or what the tooling currently outputs when building a frontend dynamic plugin.